torchgeo/datasets/ssl4eo_benchmark_landsat.py

```diff
@@ -0,0 +1,348 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""Self-Supervised Learning for Earth Observation Landsat Benchmark Evaluation."""
+
+import glob
+import os
+from typing import Callable, Optional
+
+import matplotlib.pyplot as plt
+import numpy as np
+import rasterio
+import torch
+from torch import Tensor
+
+from .geo import NonGeoDataset
+from .utils import download_url, extract_archive
+
+
+class SSL4EOLBenchmark(NonGeoDataset):
+    """SSL4EO Landsat Benchmark Evaluation Dataset.
+
+    Dataset is intended to be used for evaluation of SSL techniques. Each
+    benchmark dataset consists of 25,000 images with corresponding land
+    cover classification masks.
```

Comment:
Should mention number of classes in one of the below bullet points. I forget how many there are for CDL but it's ~100 I think. I guess you'll figure that out once you finish the mapping code.